### PR TITLE
Issue9, 17, 24, 29, 31

### PR DESCRIPTION
**Pull request by WendyW09**
## Change Summary 📝

#issue9
- Add visualisation for user workload on dashboard page, allow it to interact with the database and display the information of the currently logged in user.

#issue17

- Change the entry of the edit page from the view page.

- Modify the navigation bar of the edit page to match the view page.

**Add edit allocation detail function on edit page.**

- HoS, HoD, Admin can modify the assigned hours and send comments, the database will be updated at the same time.

- The adjusted hours will be overwrite the assigned hours on database.

- Add calculate function for adjusted workload point=adjusted hours/contract hours.

#issue24

- Add send comments function on edit page and the database will be updated when the Comment successful.

#issue29

- Add permission for different user (HoS, HoDs, Admin) can modify the assigned hours and send comments, Staff only can send 

#issue31

- Modified the previous formatting issues of the navigation bar of the entire system, and the formatting issues of the footer.

---

## Change Form 📋
Please mark as NA if not applicable. If a certain criterion is not met, kindly provide a reason.

- [x] The pull request title has an issue number
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [ ] Relevant tests (if applicable) have been added or updated.

---

## Any background context you want to provide? 🖼️

Any additional context or screenshots that could help the reviewers?

## Reviewers 🕵️

(Optional) Suggest reviewers that might be interested in this PR.

### CODE REVIEW

File: app/templates/edit_allocation_detail.html [hunk 1]
```diff
@@ -1,180 +1,304 @@
...
+                url: '/set_workload_allocation',
+                data: post,
+                success: function (res) {
+                    alert(res.msg);
+                }
+            })
+        })
+
+
+    </script>
```
Comment (yylocky) [inline app/templates/edit_allocation_detail.html:271]: Are we missing the "send comment" function somewhere?

**Review by yylocky — COMMENTED** [review]
issue 9,17,29 all look good, thanks:)
issue 24 "send comment" function seems not working.
issue 31 navbar comment history alert still has some display issue, but this is minor.
Suggest we can focus on fixing the "send comment" function, thx~~

File: app/templates/edit_allocation_detail.html (hunk 1 above)
Comment (WendyW09) [reply app/templates/edit_allocation_detail.html:271]: Yes, it has been added on now, please review thanks~

**Comment by kylelhk** [comment]
Some comments / suggestions:

View page:
1. Suggest to change “Edit” to “Edit / Comment” for the sake of clarity.

Edit page:
1. There should be an "Edit" button for updating the database.
2. Upon clicking the "Edit" button, there should be a pop-up window with something like "Edit successfully".
3. Users should be able to change Category to other Category, and Task Type to Task Type under other Category.
4. When a Staff user accesses Edit page, only Send Comment function should be available.
5. @yylocky Should Unit Code and Explanation be free texts instead of a dropdown menu?
6. Any validation function for user inputs inside the table (there is one for the Send Comment function)?  If not, we can document this as a potential enhancement.

Dashboard:
1. "department" => "Department"
2. Swap the position of the Category and Task Type pie charts.

Thanks.

**Comment by kylelhk** [comment]
A bug noted below:

- Logged in as 22222221

- Edit the selected workload:
Screenshot 2023-10-13 at 18 26 22
Screenshot 2023-10-13 at 18 41 41

- When I changed the Category, the Staff Number changed as well (33333333=>12345678):
Screenshot 2023-10-13 at 18 44 26

- Edit the entry by selecting the original Staff Number as below:
image

- Instead of editing the original entry (for 33333333), another workload entry (for 12345678) was edited instead:
Screenshot 2023-10-13 at 18 48 12


=> Workaround:
Disabled all fields other than the Adjusted Hours field.

**Review by kylelhk — COMMENTED** [review]
The workaround in my previous comment has not been implemented:
"Disabled all fields other than the Adjusted Hours field."

To be documented in the User Manual:
- Users are only allowed to edit the assigned hours.
- If other field(s) needs to be changed, the assigned hour should be set to zero and a new workload should be assigned.

**Comment by yylocky** [comment]
Send comment error:
image
image

Only assigned hour should be editable:
image

File: app/templates/edit_allocation_detail.html (hunk 1 above)
Comment (yylocky) [reply app/templates/edit_allocation_detail.html:271]: Others look all good, except for "comment status" : 
if a new comment is updated, the comment status should be set to "unread".

image

New comment added but status is still "read":

image

**Review by kylelhk — APPROVED** [review]
A minor CSS issue to be documented:
image

**Pull request merged**